blaze-core/src/main/scala/org/http4s/blaze/core/util/IdentityWriter.scala

```diff
@@ -78,11 +77,15 @@ private[blaze] class IdentityWriter[F[_]](size: Long, out: TailStage[ByteBuffer]
     else {
       val msg = s"Expected `Content-Length: $size` bytes, but only $total were written."
 
-      logger.warn(msg)
+      IdentityWriter.logger.warn(msg)
 
       writeBodyChunk(chunk, flush = true).flatMap(_ =>
         Future.failed(new IllegalStateException(msg))
       )(parasitic)
     }
   }
 }
+
+private[blaze] object IdentityWriter {
+  private lazy val logger = getLogger
```

Comment:
minor nitpick: outer object is lazy already, you can make this val a non-lazy one

Reply:
In my experience, you should never eagerly compute an expression in a static object if that expression contains any kind of side effect, which is the case here. 

By making it lazy, you have a chance to witness potential exceptions as they'll bubble up to your main method. Making it eager puts you at risk of a hard-to-debug deadlock that'll just keep your JVM hanging. 

Now that being said, maybe there's something about making static objects private that changes the behaviour I'm describing.

Reply:
Agree on that one with @Baccata

Reply:
From other size lazy vals deffer initialization that can introduce latency spikes especially under contention pressure at max load.

Also, they not so stable these days: https://twitter.com/SimonrDev/status/1621516492482433032

IMHO a better option would be replacing all logger instances by one in some utility object that is initialized at startup.

Reply:
Any instance loggers might benefit by moving to the companion object.  The cost would be a few femtoseconds on the static method call instead of a `private[this]` field reference, but allocations would usually make this a win.

Consolidating to one global logger would put all log statements into one category.  They still need to be instantiated with the class or companion they're relevant to.  (Or multiple put in one object with explicit names, but that would be an endless source of copy-and-paste errors.)

Reply:
To be honest, I can't really imagine a scenario where an slf4j logger initialization would result in a JVM deadlock. It would cause a fatal "exception in initializer" error in the worst case (which I never had even in extreme experiments), and it would even have a stack trace, but not a deadlock.

By the way, most of Java ecosystem actually relies on putting logger instances in a static value, not even a static method, so if we really started having such a problem with logger inits, then we are in for a very fun ride indeed. Generally, putting side-effects in even lazy vals in companion objects is not a very robust practice, but slf4j is so prevalent and battle-tested (more so than lazy vals themselves to be honest), so I doubt it will have a major impact on debugging experience.

Reply:
I'm not convinced the lazy makes a real difference either way here, but:
* `getLogger` is a macro that expands to the traditional Java call, which as @vilunov notes, is usually a private static final.
* The log4s documentation uses eager vals in objects.
* They're mostly (all?) eager in the rest of Blaze.